app/services/GooglePassService.scala

```diff
@@ -20,62 +20,145 @@ import com.google.auth.oauth2.GoogleCredentials
 import config.AppConfig
 import play.api.Logging
 import repositories.GooglePassRepoTrait
-import googlepass.GooglePassUtil
+import services.googlepass.GooglePassUtil
 
-import java.util.UUID
-import javax.inject._
+import java.io.ByteArrayInputStream
+import java.util.{Base64, Collections, UUID}
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
-class GooglePassService @Inject() (
+trait GooglePassService {
+
+  def getPassUrlByPassIdAndNINO(
+    passId: String,
+    nino: String
+  )(implicit ec: ExecutionContext): Future[Option[String]]
+
+  def getQrCodeByPassIdAndNINO(
+    passId: String,
+    nino: String
+  )(implicit ec: ExecutionContext): Future[Option[Array[Byte]]]
+
+  def createPass(
+    name: String,
+    nino: String,
+    expirationDate: String
+  )(implicit ec: ExecutionContext): Either[Exception, String]
+}
+
+@Singleton
+class StubGooglePassService @Inject() (
```

Comment:
Same here. No need to stub.
The google pass is just a url to https://pay.google.com/gp/v/save and there is no access to google needed to generate it. It will locally as is.